demosplan/DemosPlanCoreBundle/Logic/AssessmentTable/KeysAtEndSorter.php

```diff
@@ -46,13 +45,15 @@ public function sortArray(array $array): array
         if (0 === count($array)) {
             return $array;
         }
-        foreach ($this->keys as $key) {
-            if (array_key_exists($key, $array)) {
-                $toBePlacedLast = $array[$key];
-                unset($array[$key]);
-                $array[$key] = $toBePlacedLast;
-            }
-        }
+        // Sort array keys based on their position in $this->keys
+        uksort($array, function ($a, $b) {
+            $posA = array_search($a, $this->keys);
+            $posB = array_search($b, $this->keys);
+
+            // If both keys are in $this->keys, compare their positions
+            // If only one key is in $this->keys, prioritize moving it to the end
+            return ($posA && $posB) ? $posA - $posB : ($posB ? -1 : 0);
+        });
```

Comment:
It seems like this will ensure the order of keys is carried over into the array, while breaking the intended use to sort all matching keys to the very end of the array.